src/compute/src/server.rs

```diff
@@ -797,23 +818,24 @@ impl<'w, A: Allocate + 'static> Worker<'w, A> {
             // If it were broken out by `GlobalId` then we could drop only those of dataflows we drop.
             compute_state.subscribe_response_buffer = Rc::new(RefCell::new(Vec::new()));
 
-            // The controller expects the logging collections to be readable from the minimum time
-            // initially. We cannot recreate the logging arrangements without restarting the
-            // instance, but we can pad the compacted times with empty data. Doing so is sound
-            // because logging collections from different replica incarnations are considered
-            // distinct TVCs, so the controller doesn't expect any historical consistency from
-            // these collections when it reconnects to a replica.
+            // Reconcile log collections. This involves assigning them their new `GlobalId`s, in
+            // case they changed.
+            //
+            // Additionally, the controller expects the logging collections to be readable from the
+            // minimum time initially. We cannot recreate the logging arrangements without
+            // restarting the instance, but we can pad the compacted times with empty data. Doing
+            // so is sound because logging collections from different replica incarnations are
+            // considered distinct TVCs, so the controller doesn't expect any historical
+            // consistency from these collections when it reconnects to a replica.
             //
             // TODO(database-issues#8152): Consider resolving this with controller-side reconciliation instead.
-            if let Some(config) = old_instance_config {
-                for id in config.logging.index_logs.values() {
-                    let trace = compute_state
-                        .traces
-                        .remove(id)
-                        .expect("logging trace exists");
-                    let padded = trace.into_padded();
-                    compute_state.traces.set(*id, padded);
-                }
+            for (old_id, new_id) in log_ids.values() {
+                let collection = compute_state.collections.remove(old_id).expect("exists");
+                compute_state.collections.insert(*new_id, collection);
+
+                let trace = compute_state.traces.remove(old_id).expect("exists");
+                let padded_trace = trace.into_padded();
+                compute_state.traces.set(*new_id, padded_trace);
```

Comment:
This might have the chance to confuse compute logging. We seem to log the collection when we create it normally, including the introspection indexes, and things that depend on it, and now we're patching the ID. No idea what invariants now happen to be invalidated :/

Reply:
Hm, good point. This might not be a good idea after all...